startup.py

```diff
@@ -0,0 +1,331 @@
+                              versions to search for. If set to None,
+                              search for all versions. A version string
+                              is DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52"
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # Determine a list of paths to search for 3dsMax executables based
+        # on the windows registry
+        search_paths = self._get_installation_paths_from_registry()
+        exec_paths = []
+
+        if search_paths:
+            for search_path in search_paths:
+                # Construct the expected executable name for this path.
+                # If it exists, add it to the list of exec_paths to check.
+                exec_path = os.path.join(search_path, "3dsmax.exe")
+
+                if os.path.exists(exec_path):
+                    self.logger.debug("found version in default installation path %s" % exec_path)
+                    exec_paths.append(exec_path)
+
+        sw_versions = []
+        if exec_paths:
+            for exec_path in exec_paths:
+                # Check to see if the version number can be parsed from the path name.
+                path_sw_versions = [p.lower() for p in exec_path.split(os.path.sep)
+                                    if re.match("3ds max [0-9]+[.0-9]*$", p.lower()) is not None
+                                    ]
+                if path_sw_versions:
+                    # Use this sub dir to determine the default display name
+                    # and version for the SoftwareVersion to be created.
+                    default_display = path_sw_versions[0]
+                    default_version = default_display.replace("3ds max ", "")
+                    default_display = "3ds Max %s" % default_version
+                    self.logger.debug(
+                        "Resolved version '%s' from executable '%s'." %
+                        (default_version, exec_path)
+                    )
+
+                if versions and default_version not in versions:
+                    # If this version isn't in the list of requested versions, skip it.
+                    self.logger.debug("Skipping Maya version %s ..." % default_version)
+                    continue
+
+                if not self.is_version_supported(default_version):
+                    self.logger.info(
+                        "Found 3dsMax install in '%s' but only versions %s "
+                        "and above are supported" % (exec_path, self.minimum_supported_version)
+                    )
+
+                # Create a SoftwareVersion using the information from executable
+                # path(s) found in default locations.
+                self.logger.debug("Creating SoftwareVersion for executable '%s'." % exec_path)
+                sw_versions.append(SoftwareVersion(
+                    default_version,
```

Comment:
heads up that we made some breaking changes to the interface last week as part of building nuke and houdini. There is a now a different set of params:

https://github.com/shotgunsoftware/tk-core/blob/develop/zero_config/python/tank/platform/software_launcher.py#L629